src/main/java/com/github/tomakehurst/wiremock/stubbing/StubMappingJsonRecorder.java

```diff
@@ -155,9 +155,11 @@ private void writeToMappingAndBodyFile(
 
     ResponseDefinitionBuilder responseDefinitionBuilder =
         responseDefinition().withStatus(response.getStatus()).withBodyFile(bodyFileName);
-    if (response.getHeaders().size() > 0) {
-      responseDefinitionBuilder.withHeaders(
-          withoutContentEncodingAndContentLength(response.getHeaders()));
+
+    if (response.getHeaders() != null) {
```

Comment:
I would keep the size check here too

Reply:
My memory is fuzzy, but I think there's no harm in preserving the size check too.  I must have changed this as either my code changes made it possible for this to be null or I realized it was possible to be null even without my changes.